experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -1,184 +1,187 @@
+            buildLogicalPlanFacet(queryExecution.logical()),
+            buildPhysicalPlanFacet(queryExecution.executedPlan())))
+        .job(buildJob())
+        .eventTime(ZonedDateTime.now())
+        .eventType(getEventType(jobEnd.jobResult()))
+        .producer("spark")
+        .build();
+
+    marquezContext.emit(event);
   }
 
-  private void print(String prefix, SparkPlan executedPlan) {
-    extraInfo.append(prefix + executedPlan.treeString(true, true)).append("\n");
+  protected String getEventType(JobResult jobResult) {
+    if (jobResult.getClass().getSimpleName().startsWith("JobSucceeded")){
+      return "COMPLETE";
+    }
+    return "FAIL";
   }
 
-  private void print(String prefix, LogicalPlan plan) {
-    String extra = "";
-    if (plan instanceof LogicalRelation) {
-      LogicalRelation lr = (LogicalRelation) plan;
+  private List<LineageDataset> buildInputs(LogicalPlan logical) {
+    List<LineageDataset> inputDatasets = new ArrayList<>();
+    Collection<LogicalPlan> leaves = asJavaCollection(logical.collectLeaves());
+    for (LogicalPlan leaf : leaves) {
+      if (!(leaf instanceof LogicalRelation)) {
+        continue;
+      }
+      LogicalRelation lr = (LogicalRelation) leaf;
       BaseRelation lrr = lr.relation();
-      extra = "||| " + lrr + " " + lr.output() + " " + lr.catalogTable() + " " + lr.isStreaming();
-      if (lrr instanceof HadoopFsRelation) {
-        FileIndex location = ((HadoopFsRelation)lrr).location();
-        extra += "|||" + Arrays.toString(location.inputFiles());
+      if (!(lrr instanceof HadoopFsRelation)) {
+        continue;
+      }
+      FileIndex location = ((HadoopFsRelation)lrr).location();
+      Collection<Path> rootPaths = asJavaCollection(location.rootPaths());
+      for (Path rootPath : rootPaths) {
+        LineageDataset.LineageDatasetBuilder lineageDataset = LineageDataset.builder()
+            .namespace(marquezContext.getJobNamespace())
+            .name(rootPath.toUri().toString().replaceAll(":", "_"));
+        inputDatasets.add(lineageDataset.build());
       }
     }
-    extraInfo.append(prefix + plan.getClass().getSimpleName() + ":" + extra + " plan: " + plan).append("\n");
-    Collection<LogicalPlan> children = JavaConversions.asJavaCollection(plan.children());
-    for (LogicalPlan child : children) {
-      print(prefix + "  ", child);
-    }
+
+    return inputDatasets;
   }
 
-  public void end(SparkListenerSQLExecutionEnd endEvent) {
-    logger.info("Spark sql execution ended " + endEvent);
-    if (success == null) {
-      extraInfo.append("unknown end status\n");
-    } else if (success) {
-      marquezContext.success(runId, jobId, endEvent.time(), outputs);
-    } else  {
-      marquezContext.failure(runId, jobId, endEvent.time(), outputs, error );
+  private List<LineageDataset> buildOutputs(LogicalPlan logical) {
+    List<LineageDataset> inputDatasets = new ArrayList<>();
+    if (logical instanceof InsertIntoHadoopFsRelationCommand) {
```

Comment:
same comment for unknown Outputs. We'd want to track a string representation of an output we don't understand.